cfme/tests/automate/test_git_import.py

```diff
@@ -78,3 +78,69 @@ def test_automate_git_domain_displayed_in_service(appliance, imported_domain, br
     view.modal.apply.click()
     assert view.provisioning_entry_point.value == ("/{}/Service/Provisioning/StateMachines/"
         "ServiceProvision_Template/CatalogItemInitialization".format(imported_domain.name))
+
+
+@pytest.mark.tier(2)
+@pytest.mark.parametrize(
+    ("url", "param_type", "param_value", "verify_ssl"),
+    [
+        (
+            'https://github.com/ramrexx/CloudForms_Essentials.git',
+            'branch',
+            'origin/cf4.1',
+            True
+        ),
+        (
+            'https://github.com/RedHatQE/ManageIQ-automate-git.git',
+            'tag',
+            '0.1',
+            False
+        ),
+        (
+            "https://github.com/RedHatQE/ManageIQ-automate-git.git",
+            "branch",
+            "origin/master",
+            False,
+        ),
+        (
+            "https://github.com/ganeshhubale/ManageIQ-automate-git.git",
+            "branch",
+            "origin/test",
+            False,
+        ),
+    ],
+    ids=["with_branch", "with_tag", "with_top_level_domain", "without_top_level_domain"],
+)
+def test_domain_import_git(
+    request, appliance, url, param_type, param_value, verify_ssl
+):
+    """This test case Verifies that a domain can be imported from git and Importing domain from git
+       should work with or without the top level domain directory.
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/6h
+        caseimportance: medium
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.7
+        casecomponent: Automate
+        tags: automate
+        title: Test automate git domain import top level directory
+        testSteps:
+            1. Enable server role: git Repositories Owner
+            2. Navigate to Automation > Automate > Import/Export
+            3. Create a Git Repository with the contents of a domain directory without including
+               the domain directory.
+        expectedResults:
+            1.
+            2.
+            3. Import should work with or without the top level domain directory.
+
+    Bugzilla:
+        1389823
+    """
+    repo = AutomateGitRepository(url=url, verify_ssl=verify_ssl, appliance=appliance)
+    domain = repo.import_domain_from(**{param_type: param_value})
+    request.addfinalizer(domain.delete)
```

Comment:
Question: Why not use `delete_if_exists` here?

Reply:
Thanks for this question @john-dupuy I was expecting this one.
- So if you see I am asserting domain for it's existence.(```assert domain.exists```). In case, If domain exists then test gets passed and ```delete``` will remove the domain.
- In case, If domain does not exists(**which means test gets failed**) then there is no need of teardown.
- Actually ```delete_if_exists``` is first checking the domain exists or not then it calls ```delete``` method.
- I hope you are agree with me :)